sdk/protocol-manager-poc/src/implementation/makerPlugin.ts

```diff
@@ -0,0 +1,221 @@
+import { BigNumber } from "bignumber.js";
+import { stringToHex } from "viem";
+import { z } from "zod";
+import {
+    ChainId,
+    CreateProtocolPlugin,
+    IPositionId, MakerContracts,
+    ProtocolManagerContext,
+    ProtocolPlugin
+} from "../interfaces";
+import { Percentage, TokenAmount, TokenSymbol, Price, RiskRatio } from "@summerfi/sdk-common/common"
+import type { MakerLendingPool, MakerPoolId } from "@summerfi/sdk-common/protocols"
+import { PoolType, ProtocolName, ILKType } from "@summerfi/sdk-common/protocols"
+import { Address, Position } from "@summerfi/sdk-common/common"
+import { PRESISION_BI , PRESISION } from './constants'
+
+function amountFromWei(amount: bigint): BigNumber {
+    return new BigNumber(amount.toString()).div(new BigNumber(10).pow(PRESISION.WAD))
+}
+
+function amountFromRay(amount: bigint): BigNumber {
+    return new BigNumber(amount.toString()).div(new BigNumber(10).pow(PRESISION.RAY))
+}
+
+function amountFromRad(amount: bigint): BigNumber {
+    return new BigNumber(amount.toString()).div(new BigNumber(10).pow(PRESISION.RAD))
+}
+
+export const createMakerPlugin: CreateProtocolPlugin<MakerPoolId> = (ctx: ProtocolManagerContext): ProtocolPlugin<MakerPoolId> => {
+    const plugin: ProtocolPlugin<MakerPoolId> = {
+        supportedChains: [ChainId.Mainnet],
+        getPool: async (makerPoolId: unknown): Promise<MakerLendingPool> => {
+            plugin._validate(makerPoolId)
+            const ilk = makerPoolId.ilkType
+            if (!ilk) throw new Error('Ilk type on poolId not recognised')
```

Comment:
could be moved to a validation fn as it checks the internal implementation details of sparkPoolId